run.sh

```diff
@@ -12,21 +12,89 @@ if [[ $(basename "${CONDA_PREFIX}") != "code_as_data" ]]; then
   exit 1
 fi
 
+###############################################################################
+
+# Helper functions
+
+function yaml_value {
+    cat "${1}" | shyaml get-value "${2}"
+}
+
+###############################################################################
+
+# Input data for the project
+task_view_url="$(yaml_value "${config_file}" "task_view_url")"
+
+# Files constructed by the called scripts
+cran_table="$(yaml_value "${config_file}" "cran_details_file")"
+repo_details="$(yaml_value "${config_file}" "repo_details_file")"
+dupree_benchmarks="$(yaml_value "${config_file}" "all_pkg_benchmarks_file")"
+cloc_table="$(yaml_value "${config_file}" "all_pkg_cloc_file")"
+gitsum_table="$(yaml_value "${config_file}" "all_pkg_gitsum_file")"
+
+# Parent directories for package-specific:
+# - local-storage of the github repos
+repo_dir="$(yaml_value "${config_file}" "repo_dir")"
+# - results
+pkg_results_dir="$(yaml_value "${config_file}" "pkg_results_dir")"
+
+###############################################################################
+
 # Run each step of the analysis:
 
-Rscript "scripts/01-get-devtools-cran-table.R" --config "${config_file}"
+Rscript "scripts/01-get-devtools-cran-table.R" \
+  --config "${config_file}" \
+  --url "${task_view_url}" \
+  --output "${cran_table}"
+
+Rscript "scripts/02-github-details.R" \
+  --input "${cran_table}" \
+  --repo_dir "${repo_dir}" \
+  --output "${repo_details}"
+
+# Read each line from a header-stripped 3-column .tsv
+# For each row, store the entries in 'package', 'remote_repo' and 'local_repo'
+
+while IFS=$'\t' read package remote_repo local_repo
+do
+  Rscript "scripts/03-github-clone.R" \
+    --remote_repo "${remote_repo}" \
+    --local_repo "${local_repo}"
+
+  Rscript "scripts/04a-dupree-analysis.R" \
+    --config "${config_file}" \
+    --local_repo "${local_repo}" \
+    --output_dir "${pkg_results_dir}/${package}"
+
+  Rscript "scripts/05a-cloc-analysis.R" \
+    --local_repo "${local_repo}" \
+    --package_name "${package}" \
+    --output "${pkg_results_dir}/${package}/cloc_by_file.tsv"
+
+  Rscript "scripts/06a-gitsum-analysis.R" \
+    --local_repo "${local_repo}" \
+    --package_name "${package}" \
+    --output "${pkg_results_dir}/${package}/gitsum.tsv"
 
-Rscript "scripts/02-github-details.R" --config "${config_file}"
+done < <(tail -n+2 "${repo_details}")
 
-Rscript "scripts/03-github-clone.R" --config "${config_file}"
+# Combine the package-specific results into all-package combined datasets
 
-Rscript "scripts/04-dupree-analysis.R" --config "${config_file}"
+# TODO: rewrite as `Rscript <script> --output <out_file> [any-number-of-input-files]`
```

Comment:
Make a file containing all the filepaths for dupree results; a file with all the cloc filepaths etc here; and pass that into the collapsing scripts

Reply:
Done